core/arch/x86/mangle.c

```diff
@@ -1282,21 +1282,29 @@ mangle_direct_call(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
                    instr_t *next_instr, bool mangle_calls, uint flags)
 {
     ptr_uint_t retaddr;
+#ifdef CHECK_RETURNS_SSE2
     app_pc target = NULL;
+#endif
     opnd_t pushop = instr_get_dst(instr, 1);
     opnd_size_t pushsz = stack_entry_size(instr, opnd_get_size(pushop));
-    if (opnd_is_near_pc(instr_get_target(instr)))
+    if (opnd_is_near_pc(instr_get_target(instr))) {
+#ifdef CHECK_RETURNS_SSE2
         target = opnd_get_pc(instr_get_target(instr));
-    else if (opnd_is_instr(instr_get_target(instr))) {
+#endif
+    } else if (opnd_is_instr(instr_get_target(instr))) {
+#ifdef CHECK_RETURNS_SSE2
         instr_t *tgt = opnd_get_instr(instr_get_target(instr));
         /* assumption: target's raw bits are meaningful */
         target = instr_get_raw_bits(tgt);
         ASSERT(target != 0);
         /* FIXME case 6962: for far instr, we ignore the segment and
          * assume it matches current cs */
+#endif
     } else if (opnd_is_far_pc(instr_get_target(instr))) {
+#ifdef CHECK_RETURNS_SSE2
         target = opnd_get_pc(instr_get_target(instr));
         /* FIXME case 6962: we ignore the segment and assume it matches current cs */
+#endif
```

Comment:
Can we instead move this whole if..else chain inside the existing ifdef below, along with the declaration of the target variable?

Reply:
Done.